### ISSUE TITLE
Broken Node Deployment with new `adapter-node` version `5.0.0`

### ISSUE THREAD

**Issue by LukeSchlangen** [sveltejs/kit#11959]
### Describe the bug

I am trying to use the `adapter-node`. It was working earlier today, but is failing now after the changes to the `adapter-node`. I think it has to do with this change:

https://github.com/sveltejs/kit/commit/63c7f79cb2370753c352412154559116a44c112c#diff-a177798f9cb7b67dd12d679830cd252d752b7ff6f663010dae226d59160e9021R57

### Reproduction

1. `npm create svelte@latest sveltekit-app`
2. `cd sveltekit-app`
3. `npm i -D @sveltejs/adapter-node`
4. `npm uninstall -D @sveltejs/adapter-auto`
5. Change `auto` to `node` in svelte.config.js
6. `npm run build`
7. `node build`

Error:

```
> sveltekit-app@0.0.1 start
> node build

file:///home/luke/sveltekit-app/build/index.js:285
server.server.on(
              ^

TypeError: Cannot read properties of undefined (reading 'on')
    at file:///home/luke/sveltekit-app/build/index.js:285:15

Node.js v20.11.0
```

### Logs

```Shell
luke$ npm run build

> sveltekit-app@0.0.1 build
> vite build

vite v5.1.5 building SSR bundle for production...
"confetti" is imported from external module "@neoconfetti/svelte" but never used in "src/routes/sverdle/+page.svelte".
✓ 109 modules transformed.
vite v5.1.5 building for production...
✓ 92 modules transformed.
.svelte-kit/output/client/_app/version.json                                                         0.03 kB │ gzip:  0.05 kB
.svelte-kit/output/client/.vite/manifest.json                                                       6.80 kB │ gzip:  0.94 kB
.svelte-kit/output/client/_app/immutable/assets/fira-mono-greek-ext-400-normal.CsqI23CO.woff2       7.51 kB
.svelte-kit/output/client/_app/immutable/assets/fira-mono-cyrillic-400-normal.36-45Uyg.woff2        9.10 kB
.svelte-kit/output/client/_app/immutable/assets/fira-mono-greek-400-normal.C3zng6O6.woff2          10.52 kB
.svelte-kit/output/client/_app/immutable/assets/fira-mono-latin-ext-400-normal.D6XfiR-_.woff2      11.36 kB
.svelte-kit/output/client/_app/immutable/assets/fira-mono-cyrillic-ext-400-normal.B04YIrm4.woff2   15.77 kB
.svelte-kit/output/client/_app/immutable/assets/fira-mono-latin-400-normal.DKjLVgQi.woff2          16.28 kB
.svelte-kit/output/client/_app/immutable/assets/fira-mono-all-400-normal.B2mvLtSD.woff             77.36 kB
.svelte-kit/output/client/_app/immutable/assets/svelte-welcome.0pIiHnVF.webp                      115.47 kB
.svelte-kit/output/client/_app/immutable/assets/svelte-welcome.VNiyy3gC.png                       360.81 kB
.svelte-kit/output/client/_app/immutable/assets/5.CU6psp88.css                                      0.80 kB │ gzip:  0.35 kB
.svelte-kit/output/client/_app/immutable/assets/2.Cs8KR-Bb.css                                      1.47 kB │ gzip:  0.53 kB
.svelte-kit/output/client/_app/immutable/assets/4.DOkkq0IA.css                                      3.78 kB │ gzip:  1.06 kB
.svelte-kit/output/client/_app/immutable/assets/0.CT0x_Q5c.css                                      5.15 kB │ gzip:  1.66 kB
.svelte-kit/output/client/_app/immutable/chunks/index.R8ovVqwX.js                                   0.03 kB │ gzip:  0.05 kB
.svelte-kit/output/client/_app/immutable/entry/start.B9JTtzgM.js                                    0.07 kB │ gzip:  0.08 kB
.svelte-kit/output/client/_app/immutable/chunks/stores.B6FDz9R9.js                                  0.23 kB │ gzip:  0.17 kB
.svelte-kit/output/client/_app/immutable/chunks/index.Ice1EKvx.js                                   0.51 kB │ gzip:  0.34 kB
.svelte-kit/output/client/_app/immutable/nodes/1.DC8E7nRN.js                                        0.84 kB │ gzip:  0.52 kB
.svelte-kit/output/client/_app/immutable/nodes/3.BOaWnSKH.js                                        1.57 kB │ gzip:  0.96 kB
.svelte-kit/output/client/_app/immutable/chunks/scheduler.Dk-snqIU.js                               2.25 kB │ gzip:  1.05 kB
.svelte-kit/output/client/_app/immutable/nodes/5.Dqxx0lAG.js                                        2.35 kB │ gzip:  1.11 kB
.svelte-kit/output/client/_app/immutable/nodes/2.yKhWQcH4.js                                        5.48 kB │ gzip:  2.60 kB
.svelte-kit/output/client/_app/immutable/chunks/index.hN5g6nR9.js                                   6.10 kB │ gzip:  2.58 kB
.svelte-kit/output/client/_app/immutable/entry/app.aKpoXH17.js                                      6.58 kB │ gzip:  2.61 kB
.svelte-kit/output/client/_app/immutable/nodes/0.CtryZKGp.js                                        8.77 kB │ gzip:  3.57 kB
.svelte-kit/output/client/_app/immutable/nodes/4.BfXGfAJ9.js                                       17.09 kB │ gzip:  7.07 kB
.svelte-kit/output/client/_app/immutable/chunks/entry.Cn5jEyri.js                                  27.43 kB │ gzip: 10.80 kB
✓ built in 1.18s
.svelte-kit/output/server/.vite/manifest.json                                                       6.43 kB
.svelte-kit/output/server/_app/immutable/assets/fira-mono-greek-ext-400-normal.CsqI23CO.woff2       7.51 kB
.svelte-kit/output/server/_app/immutable/assets/fira-mono-cyrillic-400-normal.36-45Uyg.woff2        9.10 kB
.svelte-kit/output/server/_app/immutable/assets/fira-mono-greek-400-normal.C3zng6O6.woff2          10.52 kB
.svelte-kit/output/server/_app/immutable/assets/fira-mono-latin-ext-400-normal.D6XfiR-_.woff2      11.36 kB
.svelte-kit/output/server/_app/immutable/assets/fira-mono-cyrillic-ext-400-normal.B04YIrm4.woff2   15.77 kB
.svelte-kit/output/server/_app/immutable/assets/fira-mono-latin-400-normal.DKjLVgQi.woff2          16.28 kB
.svelte-kit/output/server/_app/immutable/assets/fira-mono-all-400-normal.B2mvLtSD.woff             77.36 kB
.svelte-kit/output/server/_app/immutable/assets/svelte-welcome.0pIiHnVF.webp                      115.47 kB
.svelte-kit/output/server/_app/immutable/assets/svelte-welcome.VNiyy3gC.png                       360.81 kB
.svelte-kit/output/server/_app/immutable/assets/_page.CU6psp88.css                                  0.80 kB
.svelte-kit/output/server/_app/immutable/assets/_page.BBloWWo2.css                                  1.45 kB
.svelte-kit/output/server/_app/immutable/assets/_page.DOkkq0IA.css                                  3.78 kB
.svelte-kit/output/server/_app/immutable/assets/_layout.c5uvFHuT.css                                5.40 kB
.svelte-kit/output/server/chunks/prod-ssr.js                                                        0.04 kB
.svelte-kit/output/server/entries/pages/_page.js                                                    0.05 kB
.svelte-kit/output/server/chunks/index3.js                                                          0.08 kB
.svelte-kit/output/server/entries/pages/about/_page.js                                              0.13 kB
.svelte-kit/output/server/entries/pages/sverdle/how-to-play/_page.js                                0.13 kB
.svelte-kit/output/server/chunks/client.js                                                          0.28 kB
.svelte-kit/output/server/internal.js                                                               0.31 kB
.svelte-kit/output/server/entries/fallbacks/error.svelte.js                                         0.47 kB
.svelte-kit/output/server/chunks/stores.js                                                          0.54 kB
.svelte-kit/output/server/entries/pages/about/_page.svelte.js                                       1.07 kB
.svelte-kit/output/server/chunks/index2.js                                                          1.33 kB
.svelte-kit/output/server/chunks/index.js                                                           2.10 kB
.svelte-kit/output/server/entries/pages/sverdle/how-to-play/_page.svelte.js                         2.76 kB
.svelte-kit/output/server/chunks/ssr.js                                                             4.04 kB
.svelte-kit/output/server/chunks/exports.js                                                         5.96 kB
.svelte-kit/output/server/chunks/internal.js                                                        6.03 kB
.svelte-kit/output/server/entries/pages/_page.svelte.js                                             7.97 kB
.svelte-kit/output/server/entries/pages/_layout.svelte.js                                           8.49 kB
.svelte-kit/output/server/entries/pages/sverdle/_page.svelte.js                                     8.60 kB
.svelte-kit/output/server/index.js                                                                 90.21 kB
.svelte-kit/output/server/entries/pages/sverdle/_page.server.js                                   146.23 kB
✓ built in 2.99s

Run npm run preview to preview your production build locally.

> Using @sveltejs/adapter-node
  ✔ done
luke$ npm start

> sveltekit-app@0.0.1 start
> node build

file:///home/luke/sveltekit-app/build/index.js:285
server.server.on(
              ^

TypeError: Cannot read properties of undefined (reading 'on')
    at file:///home/luke/sveltekit-app/build/index.js:285:15

Node.js v20.11.0
```
```


### System Info

```Shell
System:
    OS: Linux 6.1 Debian GNU/Linux 11 (bullseye) 11 (bullseye)
    CPU: (4) x64 Intel(R) Xeon(R) CPU @ 2.20GHz
    Memory: 14.89 GB / 15.63 GB
    Container: Yes
    Shell: 5.1.4 - /bin/bash
  Binaries:
    Node: 20.11.0 - /usr/local/nvm/versions/node/v20.11.0/bin/node
    Yarn: 1.22.21 - /usr/local/nvm/versions/node/v20.11.0/bin/yarn
    npm: 10.4.0 - /usr/local/nvm/versions/node/v20.11.0/bin/npm
  npmPackages:
    @sveltejs/adapter-node: ^5.0.0 => 5.0.0 
    @sveltejs/kit: ^2.0.0 => 2.5.3 
    @sveltejs/vite-plugin-svelte: ^3.0.0 => 3.0.2 
    svelte: ^4.2.7 => 4.2.12 
    vite: ^5.0.3 => 5.1.5
```

### Severity

blocking an upgrade

**Comment by Conduitry** [comment]
Apologies for this! It should be fixed in v5.0.1 of the adapter.

**Comment by LukeSchlangen** [comment]
Was able to test and confirmed it's now working. Thank you for fixing this so quickly!
